src/Main/Tasks.cpp

```diff
@@ -89,8 +102,26 @@ auto synthStrToOpt = [](const QString& str) -> FOEDAG::Compiler::SynthesisOpt {
   return val;
 };
 
-// This will grab Synthesis related options from Compiler::SynthOpt &
-// Compiler::SynthMoreOpt, convert/combine them, and return them as an
+auto pinOptToStr = [](FOEDAG::Compiler::PinAssignOpt opt) -> QString {
+  return pinOptMap[opt];
+};
+
+auto pinStrToOpt = [](const QString& str) -> FOEDAG::Compiler::PinAssignOpt {
+  auto it = find_if(
+      pinOptMap.begin(), pinOptMap.end(),
+      [str](const std::pair<FOEDAG::Compiler::PinAssignOpt, const char*> p) {
+        return p.second == str;
```

Comment:
are you sure this works? p.second just a pointer

Reply:
Yeah it returns the option selected.